tests/common/quantization/test_passes.py

```diff
@@ -0,0 +1,192 @@
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from copy import deepcopy
+from dataclasses import dataclass
+from typing import Dict
+
+import pytest
+
+from nncf.common.graph import NNCFGraph
+from nncf.quantization.passes import filter_constant_nodes
+from nncf.quantization.passes import remove_shapeof_subgraphs
+from tests.common.quantization.entities import Edge
+from tests.common.quantization.entities import Graph
+from tests.common.quantization.entities import Node
+from tests.common.quantization.metatypes import READ_VARIABLE_METATYPES
+from tests.common.quantization.metatypes import SHAPEOF_METATYPES
+from tests.common.quantization.mock_graphs import create_nncf_graph
+from tests.shared.nx_graph import compare_nx_graph_with_reference
+from tests.shared.paths import TEST_ROOT
+
+REFERENCES_PATH = TEST_ROOT / "common" / "data"
+
+
+@dataclass
+class TestCaseData:
+    name: str
+    methods_with_kwargs: Dict[callable, Dict]
+
+
+def graph_with_shapeof() -> NNCFGraph:
+    graph = Graph(
+        [
+            Node(0, "parameter"),
+            Node(1, "power"),
+            Node(2, "constant"),
+            Node(3, "multiply"),
+            Node(4, "constant"),
+            Node(5, "shapeof"),
+            Node(6, "interpolate"),
+            Node(7, "strided_slice"),
+            Node(8, "constant"),
+            Node(9, "constant"),
+            Node(10, "constant"),
+            Node(11, "concat"),
+            Node(12, "constant"),
+            Node(13, "convert"),
+            Node(14, "convert"),
+            Node(15, "divide"),
+            Node(16, "constant"),
+            Node(17, "add"),
+            Node(18, "add"),
+            Node(19, "read_value"),
+            Node(20, "constant"),
+        ],
+        [
+            Edge(0, 0, 1, 0),
+            Edge(2, 0, 1, 1),
+            Edge(1, 0, 3, 0),
+            Edge(4, 0, 3, 1),
+            Edge(3, 0, 6, 0),
+            Edge(3, 0, 5, 0),
+            Edge(5, 0, 7, 0),
+            Edge(8, 0, 7, 1),
+            Edge(9, 0, 7, 2),
+            Edge(10, 0, 7, 3),
+            Edge(7, 0, 11, 0),
+            Edge(12, 0, 11, 1),
+            Edge(11, 0, 6, 1),
+            Edge(11, 0, 13, 0),
+            Edge(13, 0, 15, 0),
+            Edge(15, 0, 17, 0),
+            Edge(16, 0, 17, 1),
+            Edge(17, 0, 6, 2),
+            Edge(5, 0, 14, 0),
+            Edge(14, 0, 15, 1),
+            Edge(6, 0, 18, 0),
+            Edge(20, 0, 19, 0),
+            Edge(19, 0, 18, 1),
+        ],
+    )
+    return create_nncf_graph(graph)
+
+
+def graph_with_constants() -> NNCFGraph:
+    graph = Graph(
```

Comment:
This is really hard to understand the graph. Maybe you can save it in .dot format and then load, or, probably, add here a visualization.

Reply:
Do we have any mechanisms to restore the `NNCFGraph` instance from the .dot file?

Reply:
I am not aware of such mechanisms.